app/services/container_topology_service.rb

```diff
@@ -117,35 +117,30 @@ def build_entity_data(entity)
     data
   end
 
-  def entity_status(entity, kind)
-    case kind
-    when 'Vm', 'Host' then entity.power_state.capitalize
-    when 'ContainerNode'
-      ready_status = 'Unknown'
+  def entity_status(entity)
+    if entity.kind_of?(Host) || entity.kind_of?(Vm)
+      status = entity.power_state.capitalize
+    elsif entity.kind_of?(ContainerNode)
+      status = 'Unknown'
       entity.container_conditions.each do |condition|
         if condition.try(:name) == 'Ready' && condition.try(:status) == 'True'
-          ready_status = 'Ready'
+          status = condition.name
         else
-          ready_status = 'NotReady'
+          status = 'NotReady'
```

Comment:
So, is this always NotReady if the last one is NotReady?  What about the opposite?  Note, this is how it was before, just trying to understand what's going on here.